### PR TITLE
Add "legacy" themes

### LINKED ISSUE #14858: Plan for new default theme
- [ ] rename the current "light" / "dark" -> "Light Seamless" / "Dark Seamless"
- [ ] re-add the 1.15-like light/dark as "light" and "dark"
- [ ] remove "terminalBackground" from "system"
- [ ] Change userDefaults theme to "Dark Seamless"
- [ ] revert default.json to "system"

Yes, because a lot of people are about to be affected by this.

### References
#14844

### PR DESCRIPTION
**Pull request by zadjii-msft**
This is a minimal version of the requests for #14858. In that thread we discussed FULL reverting the default themes to the old ones. In later discussion, we decided _meh_, let's just ship the legacy themes too, so it's easy to go back if you should choose. The default still remains the sane `dark`, but the `legacy*` themes are all right there, and given the same special treatment as the other inbox themes.

Closes #14858
Closes #14844

### CODE REVIEW

**Comment by zadjii-msft** [comment]
New themes (fresh settings.json)
image

"Light (Legacy)"
image

"Light"
image

File: src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp [hunk 1]
```diff
@@ -954,9 +958,9 @@ void CascadiaSettings::_researchOnLoad()
         // dark: 2
         // a custom theme: 3
         const auto themeChoice = themeInUse == L"system" ? 0 :
-                                 themeInUse == L"light"  ? 1 :
-                                 themeInUse == L"dark"   ? 2 :
-                                                           3;
+                                                           themeInUse == L"light" ? 1 :
```
Comment (DHowett) [inline src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp:961]: *ack*

File: src/cascadia/TerminalSettingsModel/defaults.json [hunk 2]
```diff
@@ -316,6 +316,38 @@
...
+            "window":
+            {
+                "applicationTheme": "light"
+            }
+        },
+        {
+            "name": "legacySystem",
+            "tab":
+            {
+                "background": null,
```
Comment (DHowett) [inline src/cascadia/TerminalSettingsModel/defaults.json:348]: how does this even work? that's neat...

File: src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp (hunk 1 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp:961]: gah

**Review by carlos-zamora — APPROVED** [review]

**Review by DHowett — APPROVED** [review]

**Pull request merged**